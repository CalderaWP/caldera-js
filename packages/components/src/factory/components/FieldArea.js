import React, { Fragment } from 'react';
import { fieldAreaFactory } from '../factories/fieldAreaFactory';

export const FieldArea = ({
	field,
	onChange,
	onBlur,
	fieldErrors,
	fieldsTouch,
}) => (
	<Fragment>
		{fieldAreaFactory(field, onChange, onBlur, fieldErrors, fieldsTouch)}
	</Fragment>
);
