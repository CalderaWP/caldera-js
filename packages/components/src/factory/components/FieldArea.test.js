import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import {
	checkboxFieldset,
	selectField,
	checkboxField,
	numberField,
	textField,
	radioField,
	emailField
} from '../fields.fixtures';

import { FieldArea } from './FieldArea';

describe('FieldArea component', () => {
	let onChange;
	let onBlur;
	beforeEach(() => {
		onChange = jest.fn();
		onBlur = jest.fn();
	});

	it('Creates a text field', () => {
		const component = renderer.create(
			<FieldArea field={textField} onChange={onChange} onBlur={onBlur} />
		);
		expect(component.toJSON()).toMatchSnapshot();
	});
	it('Shows errors', () => {
		const fieldErrors = {
			[textField.fieldId]: 'Fail Cake!'
		};
		const fieldsTouched = {
			[textField.fieldId]: true
		};

		const component = renderer.create(
			<FieldArea
				field={textField}
				onChange={onChange}
				onBlur={onBlur}
				fieldErrors={fieldErrors}
				fieldsTouch={fieldsTouched}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Creates an select field', () => {
		const component = renderer.create(
			<FieldArea
				field={selectField}
				onChange={onChange}
				onBlur={onBlur}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Changes calls change handler of radio field', () => {
		const component = mount(
			<FieldArea field={radioField} onChange={onChange} onBlur={onBlur} />
		);

		component
			.find('input')
			.first()
			.simulate('change');
		expect(onChange.mock.calls.length).toBe(1);
	});
	it('Changes calls change handler of checkbox field', () => {
		const component = mount(
			<FieldArea
				field={checkboxField}
				onChange={onChange}
				onBlur={onBlur}
			/>
		);

		component.find('input').simulate('change');
		expect(onChange.mock.calls.length).toBe(1);
	});
});
