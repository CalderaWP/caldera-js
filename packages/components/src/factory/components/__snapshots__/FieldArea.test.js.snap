// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldArea component Creates a text field 1`] = `
<div
  className="caldera-field-wrapper caldera-field-wrapper-text"
>
  <div
    className="components-base-control caldera-field caldera-field-text"
  >
    <div
      className="components-base-control__field"
    >
      <label
        className="components-base-control__label"
        htmlFor="inspector-text-control-0"
      >
        First Name
      </label>
      <input
        aria-describedby="inspector-text-control-0__help"
        className="components-text-control__input"
        id="firstName"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder={undefined}
        type="text"
        value="roy"
      />
    </div>
    <p
      className="components-base-control__help"
      id="inspector-text-control-0__help"
    >
      your first name
    </p>
  </div>
</div>
`;

exports[`FieldArea component Creates an select field 1`] = `
<div
  className="caldera-field-wrapper caldera-field-wrapper-text"
>
  <div
    className="components-base-control caldera-field caldera-field-select"
  >
    <div
      className="components-base-control__field"
    >
      <label
        className="components-base-control__label"
        htmlFor="inspector-select-control-0"
      >
        Select Field Label
      </label>
      <select
        aria-describedby="inspector-select-control-0__help"
        className="components-select-control__input"
        id="selectFieldId"
        multiple={false}
        onChange={[Function]}
        value={1}
      >
        <option
          value={0}
        >
          Zero
        </option>
        <option
          value={1}
        >
          One
        </option>
        <option
          value={2}
        >
          Two
        </option>
      </select>
    </div>
    <p
      className="components-base-control__help"
      id="inspector-select-control-0__help"
    >
      Select field description
    </p>
  </div>
</div>
`;

exports[`FieldArea component Shows errors 1`] = `
<div
  className="caldera-field-wrapper caldera-field-wrapper-text"
>
  <div
    className="components-base-control caldera-field caldera-field-text"
  >
    <div
      className="components-base-control__field"
    >
      <label
        className="components-base-control__label"
        htmlFor="inspector-text-control-1"
      >
        First Name
      </label>
      <input
        aria-describedby="inspector-text-control-1__help"
        className="components-text-control__input"
        id="firstName"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder={undefined}
        type="text"
        value="roy"
      />
    </div>
    <p
      className="components-base-control__help"
      id="inspector-text-control-1__help"
    >
      your first name
    </p>
  </div>
  <div
    className="caldera-components-message has-error caldera-components-error"
  >
    Fail Cake!
  </div>
</div>
`;
